tests/rkt_net_test.go

```diff
@@ -86,29 +86,28 @@ func TestNetHostConnectivity(t *testing.T) {
 	child := spawnOrFail(t, cmd)
 
 	ga := testutils.NewGoroutineAssistant(t)
+	ga.Add(2)
+
 	// Child opens the server
-	ga.Add(1)
 	go func() {
 		defer ga.Done()
-		waitOrFail(t, child, true)
+		ga.WaitOrFail(child)
 	}()
 
 	// Host connects to the child
-	ga.Add(1)
 	go func() {
 		defer ga.Done()
 		expectedRegex := `serving on`
 		_, out, err := expectRegexWithOutput(child, expectedRegex)
 		if err != nil {
 			ga.Fatalf("Error: %v\nOutput: %v", err, out)
-			return
 		}
 		body, err := testutils.HttpGet(httpGetAddr)
 		if err != nil {
 			ga.Fatalf("%v\n", err)
-			return
+		} else {
```

Comment:
I agree that this case is not necessary since `t.Fatalf(..)` will exit the current goroutine already, no need to call `runtime.Goexit()` either. [testing.Fatalf()](https://golang.org/pkg/testing/#B.Fatalf) calls [testing.FailNow()](https://golang.org/pkg/testing/#B.FailNow) which [calls runtime.Goexit()](https://github.com/golang/go/blob/release-branch.go1.4/src/testing/testing.go#L329) at the end.